src/bucket/BucketTests.cpp

```diff
@@ -835,6 +835,64 @@ TEST_CASE("bucket persistence over app restart", "[bucket][bucketpersist]")
     }
 }
 
+TEST_CASE("bucket count", "[bucket][count]")
+{
+    VirtualClock clock;
+    Config cfg(getTestConfig());
+    Application::pointer app = Application::create(clock, cfg);
+    BucketList& bl = app->getBucketManager().getBucketList();
+    std::vector<stellar::LedgerKey> emptySet;
+
+    for (uint32_t nLedgers = 2; nLedgers <= 256; nLedgers++)
+    {
+        CLOG(INFO, "Bucket") << "------------------ Ledgers = " << nLedgers << " ------------------";
+        app->getClock().crank(false);
+        auto ledgers = LedgerTestUtils::generateValidLedgerEntries(1);
+        ledgers[0].lastModifiedLedgerSeq = nLedgers;
+        bl.addBatch(*app, nLedgers, ledgers, emptySet);
+
+        for (uint32_t level = 0; level < BucketList::kNumLevels - 1; level++)
+        {
+            CLOG(INFO, "Bucket") << "Checking Curr[" << level << "]";
+            auto curr = bl.getLevel(level).getCurr();
+            std::set<uint32_t> currLedgers;
+            uint32_t currMin = std::numeric_limits<uint32_t>::max();
+            for (Bucket::InputIterator iter(curr); iter; ++iter)
+            {
+                auto lastModified = EntryFrame::FromXDR((*iter).liveEntry())->getLastModified();
```

Comment:
you don't need the `FromXDR` here

Reply:
(you can access the `lastModified` field directly)

Reply:
Fixed